### PR TITLE
7 vote on article

### PR DESCRIPTION
**Pull request by simonhighfield**
I updated the css class of the buttons in-line with a ternary statement. However before that I tried doing it with a useEffect() which was lsitening for when the buttons were pressed (so the state variable: hasVoted). I didn't use that way as I wasn't sure how I could then apply a css class to the button inside the useEffect(). My instinct would be to assign an ID to that button, so getElementById then classList.add etc. However, I don't think we can getElementByID as that is for the DOM, rather than react ...?

So basically my question is: how could I have achevied something similar inside the useEffect() ? 

Thanks :D

### CODE REVIEW

**Comment by nc-saima-khanom** [comment]
Hi Simon, using a ternary operator to dynamically set the classname of an element is pretty much the standard way of doing things! Doing it using a useEffect is quite unorthodox and would complicate this logic unnecessarily - you'd probably need to have the classname as a state and update the state in the useEffect to achieve this but as I said quite unnecessary when it can be achieved with a ternary!

File: src/components/ViewArticle.jsx [hunk 1]
```diff
@@ -26,6 +26,10 @@ export default function ViewArticle(props) {
         })
     }, [])
 
+    function handleVote() {
```
Comment (nc-saima-khanom) [inline src/components/ViewArticle.jsx]: this function just calls another function and does nothing else - consider moving the logic for voteOnArticle to here instead to remove the layer in between

File: src/components/ViewArticle.jsx [hunk 2]
```diff
@@ -26,6 +26,10 @@ export default function ViewArticle(props) {
...
+        voteOnArticle(currentArticle.article_id)
```
Comment (nc-saima-khanom) [inline src/components/ViewArticle.jsx]: consider optimistically rendering the votes here: https://notes.northcoders.com/courses/js-front-end/ui-behaviours-and-optimistic-rendering#optimistic-rendering

File: utils/handleVote.js [hunk 3]
```diff
@@ -42,16 +42,16 @@ export default function handleVote(currentArticle, voteDirection, hasVoted, setH
         setHasVoted("")
         voteOnArticle(currentArticle.article_id, 1).catch((err) => {
             setIsError(true)
-            setCurrentVotes(currentVotes + 1)
+            setCurrentVotes(currentVotes - 1)
         })
     }
 
     if (voteDirection === "down" && hasVoted === "up") {
```
Comment (nc-saima-khanom) [inline utils/handleVote.js:49]: nicely handled logic of being able to use the up/down vote options in tandem with each other!
